ast/env.go

```diff
@@ -253,6 +270,12 @@ func (n *typeTreeNode) Put(path Ref, tpe types.Type) {
 	curr.value = tpe
 }
 
+func (n *typeTreeNode) PutArg(path Ref, i int, tpe types.Type) {
+	pos := json.Number(strconv.Itoa(i))
```

Comment:
I'd rather see the type info for user functions represented explicitly in the type env as opposed to implicitly as it's currently implemented.